test/integration/generated_logs_test.go

```diff
@@ -0,0 +1,173 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+)
+
+// Verifies logs generated in the /tmp directory
+func TestGeneratedLogs(t *testing.T) {
+
+	profile := UniqueProfileName("testGenLogs")
+
+	ctx, cancel := context.WithTimeout(context.Background(), Minutes(25))
+	defer CleanupWithLogs(t, profile, cancel)
+
+	logDir := filepath.Join(os.TempDir(), profile)
+	if err := os.MkdirAll(logDir, 0755); err != nil {
+		t.Fatalf("Unable to make logDir %s: %v", logDir, err)
+	}
+	defer os.RemoveAll(logDir)
+
+	// This should likely use multi-node once it's ready
+	// use `--log_dir` flag to run isolated and avoid race condition - ie, failing to clean up (locked) log files created by other concurently-run tests, or counting them in results
+	args := append([]string{"start", "-p", profile, "-n=1", "--memory=2250", "--wait=false", fmt.Sprintf("--log_dir=%s", logDir)}, StartArgs()...)
+
+	rr, err := Run(t, exec.CommandContext(ctx, Target(), args...))
+	if err != nil {
+		t.Errorf("%q failed: %v", rr.Command(), err)
+	}
+
+	stdout := rr.Stdout.String()
+	stderr := rr.Stderr.String()
+
+	if t.Failed() {
+		t.Logf("minikube stdout:\n%s", stdout)
+		t.Logf("minikube stderr:\n%s", stderr)
+	}
+
+	logsToBeRemoved, err := filepath.Glob(filepath.Join(logDir, "minikube_*"))
+	if err != nil {
+		t.Error("failed to clean old log files", err)
+	}
+	cleanupLogFiles(t, logsToBeRemoved)
+
+	logTests := []struct {
+		command          string
+		args             []string
+		runCount         int // number of times to run command
+		expectedLogFiles int // number of logfiles expected after running command runCount times
+	}{
+		{
+			command:          "start",
+			args:             []string{"--dry-run"},
+			runCount:         175, // calling this 175 times should create 2 files with 1 greater than 1M
+			expectedLogFiles: 2,
+		},
+		{
+			command:          "status",
+			runCount:         100,
+			expectedLogFiles: 1,
+		}, {
+			command:          "pause",
+			runCount:         5,
+			expectedLogFiles: 1,
+		}, {
+			command:          "unpause",
+			runCount:         1,
+			expectedLogFiles: 1,
+		}, {
+			command:          "stop",
+			runCount:         1,
+			expectedLogFiles: 1,
+		},
+	}
+
+	for _, test := range logTests {
+		t.Run(test.command, func(t *testing.T) {
+
+			// before starting the test, ensure no other logs from the current command are written
+			logFiles, err := filepath.Glob(filepath.Join(logDir, fmt.Sprintf("minikube_%s*", test.command)))
+			if err != nil {
+				t.Errorf("failed to get old log files for command %s : %v", test.command, err)
+			}
+			cleanupLogFiles(t, logFiles)
+
+			args := []string{"-p", profile, "--log_dir", logDir, test.command}
```

Comment:
Hi , I have created a test file test/integration/generated_logs_test.go to verify the generated logs when running the minikube subcommands. Most of the functionality here was part of the file  test/integration/error_spam_test.go . Please let me know if this would be the correct approach for this. Or else should I make these changes part of the previous file

The changes in test are 

1) Remove all files that match `minikube_*` after the first start ( line 61)
2) Test a single run to see the proper filename getting generated before we run multiple runs
3) Test for #11382 by changing the args variable to include flags before the subcommand